controllers/vspheremachine_controller.go

```diff
@@ -249,14 +250,54 @@ func (r *machineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 	})
 	// always patch the VSphereMachine object
 	defer func() {
+		// Before computing ready condition, make sure that VirtualMachineProvisioned is always set.
+		// NOTE: This is required because v1beta2 conditions comply to guideline requiring conditions to be set at the
+		// first reconcile.
+		if c := v1beta2conditions.Get(machineContext.GetVSphereMachine(), infrav1.VSphereMachineVirtualMachineProvisionedV1Beta2Condition); c == nil {
+			if machineContext.GetReady() {
+				v1beta2conditions.Set(machineContext.GetVSphereMachine(), metav1.Condition{
+					Type:   infrav1.VSphereMachineVirtualMachineProvisionedV1Beta2Condition,
+					Status: metav1.ConditionTrue,
+					Reason: infrav1.VSphereMachineVirtualMachineProvisionedV1Beta2Reason,
+				})
+			} else {
+				v1beta2conditions.Set(machineContext.GetVSphereMachine(), metav1.Condition{
+					Type:   infrav1.VSphereMachineVirtualMachineProvisionedV1Beta2Condition,
+					Status: metav1.ConditionFalse,
+					Reason: infrav1.VSphereMachineVirtualMachineNotProvisionedV1Beta2Reason,
+				})
+			}
+		}
+
 		// always update the readyCondition.
 		conditions.SetSummary(machineContext.GetVSphereMachine(),
 			conditions.WithConditions(
 				infrav1.VMProvisionedCondition,
 			),
 		)
 
+		if err := v1beta2conditions.SetSummaryCondition(machineContext.GetVSphereMachine(), machineContext.GetVSphereMachine(), infrav1.VSphereMachineReadyV1Beta2Condition,
+			v1beta2conditions.ForConditionTypes{
+				infrav1.VSphereMachineVirtualMachineProvisionedV1Beta2Condition,
+			},
+			// Using a custom merge strategy to override reasons applied during merge.
+			v1beta2conditions.CustomMergeStrategy{
+				MergeStrategy: v1beta2conditions.DefaultMergeStrategy(
+					// Use custom reasons.
+					v1beta2conditions.ComputeReasonFunc(v1beta2conditions.GetDefaultComputeMergeReasonFunc(
+						infrav1.VSphereMachineNotReadyV1Beta2Reason,
```

Comment:
```suggestion
						clusterv1.PausedV1Beta2Condition,
						infrav1.VSphereMachineNotReadyV1Beta2Reason,
```

Reply:
I think the idea of this comment is to add it to owned conditions, right? (it's just not the right palce, see l.300)

Reply:
Fixed I think

Reply:
Looks good as far as I can tell. But keeping this open so Christian can verify

Reply:
ah yeah, comment is at the wrong place 😂